spring-shell2-shell1-adapter/src/main/java/org/springframework/shell2/legacy/LegacyParameterResolver.java

```diff
@@ -113,17 +112,67 @@ public ParameterDescription describe(MethodParameter parameter) {
 
 	@Override
 	public List<CompletionProposal> complete(MethodParameter parameter, CompletionContext context) {
-		return null;
+		// DevNote: in Legacy the user is either completing an argument (-- prefix) or the value of the argument (e.g. no arity to consider).
+		// Other cases to be considered include default keys and default values
+		
+		CliOption option = parameter.getParameterAnnotation(CliOption.class);
+		List<String> keys = Arrays.asList(option.key());
+		
+		int currentWordIndex = context.getWordIndex();		
+		
+		// 1) has a key for this parameter already been defined?
+		boolean keyAlreadyDefined = context.getWords().stream()
+				.filter(word -> word.startsWith(CLI_PREFIX) && keys.contains(word.substring(CLI_PREFIX.length())))
+				.map(word -> true)
+				.findFirst().orElse(false);
+		
+		if (keyAlreadyDefined) {
+			// if a key has already been provided for this parameter, then there is no need to provide completion
+			// proposals
+			return Collections.emptyList();
```

Comment:
Not sure I follow: wouldn't this case trap the `myCommand --myKey <TAB>` case? This is basically your case 3 below, but wouldn't this short circuit it?